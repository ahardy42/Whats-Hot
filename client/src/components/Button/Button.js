import React, {useState} from 'react';
import './Button.sass';

const Button = ({extraClass, type, handleClick, children}) => {
    return(
        <button type={type} className={`btn ${extraClass ? extraClass : ""}`} onClick={handleClick}>{children}</button>
    )
}

export default Button;