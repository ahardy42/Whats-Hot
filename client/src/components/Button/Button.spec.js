import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('button render test suite', () => {
    afterEach(() => {
        cleanup();
    })
    // unit test
    it('renders a button', () => {
        const {container} = render(<Button>Hey!</Button>);
        const theButton = container.firstChild;
        expect(theButton).toContainElement(theButton);
    });
    it('renders attributes based on props', () => {
        const {container} = render(<Button type="button" extraClass="dude">Hey!</Button>);
        const theButton = container.firstChild;
        expect(theButton.textContent).toEqual('Hey!');
        expect(theButton.className).toBe('btn dude');
    });
    it('responds to clicks', () => {
        const func = jest.fn(() => 'clicked')
        const {container} = render(<Button type="button" extraClass="dude" handleClick={func}>Hey!</Button>);
        const theButton = container.firstChild;
        fireEvent.click(theButton);
        expect(func).toHaveBeenCalledTimes(1);
    })
    // snapshot test
    it('renders correctly', () => {
        const tree = renderer.create(<Button>Hey!</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
