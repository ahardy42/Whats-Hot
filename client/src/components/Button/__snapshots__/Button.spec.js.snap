// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render test suite renders correctly 1`] = `
<button
  className="btn "
>
  Hey!
</button>
`;
